pkg/build/build.go

```diff
@@ -24,24 +24,47 @@ import (
 	"k8s.io/release/pkg/release"
 )
 
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+
 var DefaultExtraVersionMarkers = []string{}
 
-// Instance is the main structure for creating and pushing builds.
-type Instance struct {
+// Client is the main structure for creating and pushing builds.
+type Client struct {
```

Comment:
I feel like the cleanest thing to do here would be to just make a `Client` interface, then have a `New(opts *build.Options)` function that returns the default client, then we can have fakes of the `Client` interface if needed. For the options, I think it makes sense to think about the different "backends" that we are interacting with, which is roughly:
- a filesystem (the local fs by default)
- a remote object storage service (GCS by default)
- a remote OCI image registry (GCR by default)

These could each be interfaces with concrete implementations passed as options. For the filesystem, we can use something like `afero`. For the registry and object storage, it may make sense to implement our own pkgs with interfaces for those and think about what concrete implementations would look like. For instance, a `Registry` interface would likely need methods such as `Push` / `Pull` / etc. and a `Storage` interface may have methods such as `Copy` / `Sync` / `Delete` / etc.

We would then have concrete implementations. For the registry, a concrete implementation could be shelling out to `docker` binary, a library built on something like `go-containerregistry`, etc.. For the storage, it could be shelling out to `gsutil`, using the GCP Go SDK, using the AWS S3 SDK, etc.

Anyway, I realize this is somewhat of a large increase in scope. I would be happy to get the registry and storage pkgs created for you to use here if that is desirable :) I think building in this manner (i.e. thinking about the external services we interact with and using generic interfaces that satisfy them) is going to go a long way towards making `krel` able to be used more generally as a release tool

Reply:
If we make `Client` an interface we would still need a `defaultClient` structure here somewhere to implement that interface.

Reply:
@saschagrunert -- yep, I chatted with Dan earlier about the approach!

I'm going to collapse some of the commits on this PR and put up a no-frills implementation for y'all to review. :)